### ISSUE TITLE
network.Usage unmarshall causes error

### ISSUE THREAD

**Issue by nathantypanski** [Azure/azure-sdk-for-go#906]
I have some code that creates a [`network.UsagesClient`](https://godoc.org/github.com/Azure/azure-sdk-for-go/arm/network#UsagesClient) and calls [`ListComplete()`](https://sourcegraph.com/-/godoc/refs?def=UsagesClient%2FListComplete&pkg=github.com%2FAzure%2Fazure-sdk-for-go%2Farm%2Fnetwork&repo=github.com%2FAzure%2Fazure-sdk-for-go) on it.

This immediately results in an error unmarshalling JSON:

```
2017/12/14 16:39:01 network.UsagesClient#List: Failure responding to request: StatusCode=200 -- Original Error: Error occurred unmarshalling JSON - Error = 'json: cannot unmarshal number 0.0 into Go struct field Usage.currentValue of type int64' JSON = '{
  "value": [
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/VirtualNetworks",
      "limit": 50.0,
      "name": {
        "localizedValue": "Virtual Networks",
        "value": "VirtualNetworks"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/StaticPublicIPAddresses",
      "limit": 20.0,
      "name": {
        "localizedValue": "Static Public IP Addresses",
        "value": "StaticPublicIPAddresses"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/NetworkSecurityGroups",
      "limit": 100.0,
      "name": {
        "localizedValue": "Network Security Groups",
        "value": "NetworkSecurityGroups"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/PublicIPAddresses",
      "limit": 60.0,
      "name": {
        "localizedValue": "Public IP Addresses",
        "value": "PublicIPAddresses"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/NetworkInterfaces",
      "limit": 350.0,
      "name": {
        "localizedValue": "Network Interfaces",
        "value": "NetworkInterfaces"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/LoadBalancers",
      "limit": 100.0,
      "name": {
        "localizedValue": "Load Balancers",
        "value": "LoadBalancers"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/ApplicationGateways",
      "limit": 50.0,
      "name": {
        "localizedValue": "Application Gateways",
        "value": "ApplicationGateways"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/RouteTables",
      "limit": 100.0,
      "name": {
        "localizedValue": "Route Tables",
        "value": "RouteTables"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/RouteFilters",
      "limit": 1000.0,
      "name": {
        "localizedValue": "Route Filters",
        "value": "RouteFilters"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/NetworkWatchers",
      "limit": 1.0,
      "name": {
        "localizedValue": "Network Watchers",
        "value": "NetworkWatchers"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/PacketCaptures",
      "limit": 10.0,
      "name": {
        "localizedValue": "Packet Captures",
        "value": "PacketCaptures"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/ApplicationSecurityGroups",
      "limit": 200.0,
      "name": {
        "localizedValue": "Application Security Groups.",
        "value": "ApplicationSecurityGroups"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/DnsServersPerVirtualNetwork",
      "limit": 9.0,
      "name": {
        "localizedValue": "DNS servers per Virtual Network",
        "value": "DnsServersPerVirtualNetwork"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/SubnetsPerVirtualNetwork",
      "limit": 1000.0,
      "name": {
        "localizedValue": "Subnets per Virtual Network",
        "value": "SubnetsPerVirtualNetwork"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/IPConfigurationsPerVirtualNetwork",
      "limit": 4096.0,
      "name": {
        "localizedValue": "IP Configurations per Virtual Network",
        "value": "IPConfigurationsPerVirtualNetwork"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/PeeringsPerVirtualNetwork",
      "limit": 10.0,
      "name": {
        "localizedValue": "Peerings per Virtual Network",
        "value": "PeeringsPerVirtualNetwork"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/SecurityRulesPerNetworkSecurityGroup",
      "limit": 200.0,
      "name": {
        "localizedValue": "Security rules per Network Security Group",
        "value": "SecurityRulesPerNetworkSecurityGroup"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/SecurityRuleAddressesOrPortsPerNetworkSecurityGroup",
      "limit": 2000.0,
      "name": {
        "localizedValue": "Security rules addresses or ports per Network Security Group",
        "value": "SecurityRuleAddressesOrPortsPerNetworkSecurityGroup"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/InboundRulesPerLoadBalancer",
      "limit": 150.0,
      "name": {
        "localizedValue": "Inbound Rules per Load Balancer",
        "value": "InboundRulesPerLoadBalancer"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/FrontendIPConfigurationPerLoadBalancer",
      "limit": 10.0,
      "name": {
        "localizedValue": "Frontend IP Configurations per Load Balancer",
        "value": "FrontendIPConfigurationPerLoadBalancer"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/OutboundNatRulesPerLoadBalancer",
      "limit": 5.0,
      "name": {
        "localizedValue": "Outbound NAT Rules per Load Balancer",
        "value": "OutboundNatRulesPerLoadBalancer"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/RoutesPerRouteTable",
      "limit": 100.0,
      "name": {
        "localizedValue": "Routes per Route Table",
        "value": "RoutesPerRouteTable"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/SecondaryIPConfigurationsPerNetworkInterface",
      "limit": 256.0,
      "name": {
        "localizedValue": "Secondary IP Configurations per Network Interface",
        "value": "SecondaryIPConfigurationsPerNetworkInterface"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/InboundRulesPerNetworkInterface",
      "limit": 500.0,
      "name": {
        "localizedValue": "Inbound rules per Network Interface",
        "value": "InboundRulesPerNetworkInterface"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/RouteFilterRulesPerRouteFilter",
      "limit": 1.0,
      "name": {
        "localizedValue": "Route filter rules per Route Filter",
        "value": "RouteFilterRulesPerRouteFilter"
      },
      "unit": "Count"
    },
    {
      "currentValue": 0.0,
      "id": "/subscriptions/<subscription>/providers/Microsoft.Network/locations/eastus/usages/RouteFiltersPerExpressRouteBgpPeering",
      "limit": 1.0,
      "name": {
        "localizedValue": "Route filters per Express route BGP Peering",
        "value": "RouteFiltersPerExpressRouteBgpPeering"
      },
      "unit": "Count"
    }
  ]
}'
```

The root cause of this seems to be that [`network.Usage`](https://godoc.org/github.com/Azure/azure-sdk-for-go/arm/network#Usage) has the `Limit` and `CurrentValue` set to `int64` while the API is returning floating point values.

**Comment by vladbarosan** [comment]
Hi @nathantypanski indeed I checked the Service spec and it incorrectly specifies integer as a return type for both of this values. We will raise the issue to be corrected in the spec and once that is done it will be available in our next release.

**Comment by nathantypanski** [comment]
Looks like the spec people noticed this:

https://github.com/Azure/azure-rest-api-specs/issues/1624

Discussion here:

https://github.com/Azure/azure-rest-api-specs/pull/1603/files#r135889852

**Comment by vladbarosan** [comment]
yes. Closing as duplicate of:https://github.com/Azure/azure-rest-api-specs/issues/1624  this as duplicate since its tracked there. 

Once that gets fixed, next release will automatically have the fix.
